### ISSUE TITLE
'Parse Error: Invalid header value char' when downloading a file with name containing 'ÿ'

### ISSUE THREAD

**Issue by etiennebatise** [box/box-node-sdk#517]
- [x] I have checked that the [SDK documentation][sdk-docs] doesn't solve my issue.
- [x] I have checked that the [API documentation][api-docs] doesn't solve my issue.
- [x] I have searched the [Box Developer Forums][dev-forums] and my issue isn't already reported (or if it has been reported, I have attached a link to it, for reference).
- [x] I have searched [Issues in this repo][github-repo] and my issue isn't already reported.

### Description of the Issue
I am trying to download a file with a name containing the special character `ÿ` and the client throws an error.

For instance:
![image](https://user-images.githubusercontent.com/5644956/85149998-469d5580-b252-11ea-9951-1e164a1927ec.png)

### Steps to Reproduce
- Create a file with a name containing the character `ÿ`. 
- Get the box id of that file
- Call the following code 

```javascript
client = <insert box sdk init>
id = <insert box file id>
const tmpFile = `/tmp/${id}`
const stream = await client.files.getReadStream(id)
const out = fs.createWriteStream(tmpFile)

// Wait for download to be over
await new Promise((resolve, reject) => {
  stream.on('error', err => {
    reject(err)
  })
  stream.on('end', function () {
    resolve()
  })
  stream.pipe(out)
})

console.log(tmpFile)
```
### Expected Behavior
It should not crash.
Alternatively, it should crash with a better error. Maybe something like `Filename contains unsupported characters`.

### Error Message, Including Stack Trace

It throws an error with the following stack trace:
```
Error: Parse Error: Invalid header value char
    at TLSSocket.socketOnData (_http_client.js:479:22)
    at TLSSocket.emit (events.js:316:20)
    at addChunk (_stream_readable.js:297:12)
    at readableAddChunk (_stream_readable.js:273:9)
    at TLSSocket.Readable.push (_stream_readable.js:214:10)
    at TLSWrap.onStreamRead (internal/stream_base_commons.js:186:23)
```

I also have the following error code in my log:
```
  "code": "HPE_INVALID_HEADER_TOKEN",
```

I am pretty sure other weird characters would raise the same error.

### Versions Used
Node SDK: `1.32.0`

[sdk-docs]: ./doc
[api-docs]: https://developer.box.com/docs
[dev-forums]: https://community.box.com/t5/Platform-and-Development-Forum/bd-p/DeveloperForum
[github-repo]: https://github.com/box/box-java-sdk/search?type=Issues

**Comment by sujaygarlanka** [comment]
@etiennebatise I am not able to replicate this issue. I was able to download a file with the special character ÿ. I used the code below
```
client.files.getReadStream('file_id', null, function(error, stream) {

	if (error) {
		// handle error
	}

	// write the file to disk
	var output = fs.createWriteStream('/Users/sgarlanka/Desktop/ÿellow.pdf');
	stream.pipe(output);
});
```

**Comment by PJSimon** [comment]
Hi @etiennebatise ,

Are you still having this issue or was Sujay able to help you resolve it?  I'll close it for now, but if you still need help, please just re-open and comment!

Thanks!

**Comment by etiennebatise** [comment]
Hello again, 

I come back after a while to confirm this is still happening in `1.35`. @sujaygarlanka , I can't tell if your test has the same context. In my case, the bug happens when the file _on box_ has a `ÿ` in its name. The local file name is unimportant.

After a bit of digging, I found out issues with a similar stack trace mentioning the http-parser of nodejs. See: 
- https://github.com/nodejs/help/issues/2837
- https://github.com/nodejs/node/issues/27711
- https://github.com/nodejs/http-parser/issues/520

I can't confirm this is related but that might be a good starting point. I have tried to follow the call stack of `files.getReadStream` but I couldn't find really where the download happens.
